environment.go

```diff
@@ -338,7 +339,8 @@ func (e *envState) createGenerateTasksQueue(ctx context.Context) error {
 	e.remoteQueueGroup = remoteQueueGroup
 
 	e.closers["single-queue-generate-tasks"] = func(ctx context.Context) error {
-		e.generateTasksQueue.Runner().Close()
+		closerCtx := context.Background()
```

Comment:
I don't think you want this to block forever. 

also change the name of the closer that you register

Reply:
done